uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/performance/DeviceMonitor.java

```diff
@@ -0,0 +1,219 @@
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.performance;
+
+import android.os.Build;
+
+import androidx.test.core.app.ApplicationProvider;
+
+import com.microsoft.identity.client.ui.automation.utils.AdbShellUtils;
+
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Collects data for the current process for different {@link PerformanceProfile}. Additional profiles can included and
+ * their implementation done in {@link PerformanceProfileMonitor} then passed through the method {@link DeviceMonitor#setProfiler(PerformanceProfile, PerformanceProfileMonitor)}
+ */
+public class DeviceMonitor {
+
+    private static final String packageName = ApplicationProvider.getApplicationContext().getPackageName();
+    private static final HashMap<PerformanceProfile, PerformanceProfileMonitor<?>> profilers = new HashMap<>();
+    private static ProcessInfo processInfo = null;
+
+    static {
+        loadProcessInfo();
+        setProfiler(PerformanceProfile.CPU, new CPUMonitor());
+        setProfiler(PerformanceProfile.MEMORY, new MemoryMonitor());
+        setProfiler(PerformanceProfile.NETWORK, new NetworkUsageMonitor());
+    }
+
+    /**
+     * Get the device name string
+     *
+     * @return
+     */
+    public static String getDeviceName() {
+        String manufacturer = Build.MANUFACTURER;
+        String model = Build.MODEL;
+        if (model.toLowerCase().startsWith(manufacturer.toLowerCase())) {
+            return capitalize(model);
+        } else {
+            return capitalize(manufacturer) + " " + model;
+        }
+    }
+
+
+    private static String capitalize(String s) {
+        if (s == null || s.length() == 0) {
+            return "";
+        }
+        char first = s.charAt(0);
+        if (Character.isUpperCase(first)) {
+            return s;
+        } else {
+            return Character.toUpperCase(first) + s.substring(1);
+        }
+    }
+
+    /**
+     * Gets the current application's uid
+     *
+     * @return the application uid
+     */
+    public static int getApplicationUid() {
+        final Pattern pattern = Pattern.compile("userId=(\\d+)");
+        final String command = "dumpsys package " + packageName;
+
+        final Matcher matcher = pattern.matcher(AdbShellUtils.executeShellCommand(command));
+
+        if (matcher.find()) {
+            return Integer.parseInt(Objects.requireNonNull(matcher.group(1)));
+        }
+        return 0;
+    }
+
+    /**
+     * Get the total System memory
+     *
+     * @return
+     */
+    public static Long getTotalMemory() {
+        return processInfo.getTotalSystemMemory();
+    }
+
+    /**
+     * Get the current CPU usage
+     *
+     * @return the current cpu usage as a percentage e.g. 3.3%
```

Comment:
Is the percent sign i.e `%` included as part of the Double? I suspect not..and if not then we should fix the javadoc